### PR TITLE
feat(frontend): re-use ReceiveModal component

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

The goal is to move  the`ReceiveModal` component to lib so it can be reused by both ETH and BTC.

### CODE REVIEW

File: src/frontend/src/eth/components/receive/EthReceive.svelte [hunk 1]
```diff
@@ -23,5 +25,11 @@
 </script>
 
 <ReceiveButtonWithModal open={openReceive} isOpen={$modalEthReceive}>
-	<EthReceiveModal slot="modal" />
+	<ReceiveModal slot="modal" address={$networkAddress ?? undefined}>
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/eth/components/receive/EthReceive.svelte:28]: the `??` check is not needed here right?

File: src/frontend/src/eth/components/receive/EthReceive.svelte [hunk 2]
```diff
@@ -23,5 +25,11 @@
...
+		<svelte:fragment slot="content">
+			{#if $networkEthereum}
+				<EthReceiveMetamask />
```
Comment (AntonioVentilii-DFINITY) [inline src/frontend/src/eth/components/receive/EthReceive.svelte:31]: out of curiosity: why the `slot` definition is not here but outside the `if` condition?

Comment (DenysKarmazynDFINITY) [reply src/frontend/src/eth/components/receive/EthReceive.svelte:31]: This is required by Svelte (`Svelte: <svelte:fragment> must be a child of a component`). More discussions here - https://github.com/sveltejs/svelte/issues/5604.

File: src/frontend/src/eth/components/receive/EthReceive.svelte (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/eth/components/receive/EthReceive.svelte]: I adjusted the type of address prop so `??` is now gone.

File: src/frontend/src/eth/components/receive/EthReceive.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/eth/components/receive/EthReceive.svelte:31]: wow! thank you!

**Review by AntonioVentilii-DFINITY — APPROVED** [review]
LGTM!

**Pull request merged**